implementation/src/main/java/io/smallrye/jwt/auth/principal/KeyLocationResolver.java

```diff
@@ -79,12 +79,18 @@
     private JWTAuthContextInfo authContextInfo;
 
     public KeyLocationResolver(JWTAuthContextInfo authContextInfo) throws UnresolvableKeyException {
+        if (authContextInfo == null) {
+            throw new UnresolvableKeyException("No AuthContextInfo given");
+        }
+
         this.authContextInfo = authContextInfo;
+        LOGGER.debugf("AuthContextInfo is: %s", authContextInfo);
```

Comment:
This will call `toString` even if `debug` is not enabled ? I propose to check if debug is enabled and then log to avoid the redundant String allocation

Reply:
This should be Ok, this is the implementation of debugf: -

    public void debugf(String format, Object param1) {
        if (isEnabled(Level.DEBUG)) {
            doLogf(Level.DEBUG, FQCN, format, new Object[] { param1 }, null);
        }
    }

Reply:
@darranl Hi Darran, this is fine, I wonder, will the compiled code will effectively have something like:
```
LOGGER.debugf("AuthContextInfo is: %s", authContextInfo.toString());
```
?

Reply:
In my experience would be quite unusual to encapsulate this by ```LOGGER.isDebugEnabled()```, but of course can do so, if you wish.

Reply:
> @darranl Hi Darran, this is fine, I wonder, will the compiled code effectively have something like:
> 
> ```
> LOGGER.debugf("AuthContextInfo is: %s", authContextInfo.toString());
> ```
> 
> ?

From .class file
```java
LOGGER.debugf("AuthContextInfo is: %s", authContextInfo);
```

Reply:
@SchulteMarkus, thanks for checking, then obviously we should keep your original code :-), Darran, @darranl, thanks as well